File: backend/src/main/java/com/workflow/common/domain/PageResult.java
package com.workflow.common.domain;

import java.io.Serializable;
import java.util.List;

public class PageResult<T> implements Serializable {
    private long total;
    private int page;
    private int size;
    private List<T> rows;

    public PageResult() {}

    public PageResult(long total, int page, int size, List<T> rows) {
        this.total = total;
        this.page = page;
        this.size = size;
        this.rows = rows;
    }

    public long getTotal() { return total; }
    public void setTotal(long total) { this.total = total; }
    public int getPage() { return page; }
    public void setPage(int page) { this.page = page; }
    public int getSize() { return size; }
    public void setSize(int size) { this.size = size; }
    public List<T> getRows() { return rows; }
    public void setRows(List<T> rows) { this.rows = rows; }
}

File: backend-node/src/common/domain/page-result.ts
/**
 * 分页响应封装，逐字对齐 Java 侧 com.workflow.common.domain.PageResult。
 *
 * 字段名 total / page / size / rows 与 Java 完全一致，
 * 不可改成 pageSize / records / list 等 —— 前端直接消费这些字段。
 */
export class PageResult<T> {
  total: number
  page: number
  size: number
  rows: T[]

  constructor(total: number, page: number, size: number, rows: T[]) {
    this.total = total
    this.page = page
    this.size = size
    this.rows = rows
  }
}

export function pageResult<T>(total: number, page: number, size: number, rows: T[]): PageResult<T> {
  return new PageResult<T>(total, page, size, rows)
}
